test/bats/e2e-provider.bats

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_RESOURCE_MANIFESTS_DIR=hack/localsetup
+BATS_RESOURCE_YAML_DIR=test/bats/tests/e2e_provider
+WAIT_TIME=60
+SLEEP_TIME=1
+
+@test "secretproviderclasses crd is established" {
+  kubectl wait --for condition=established --timeout=60s crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+
+  run kubectl get crd/secretproviderclasses.secrets-store.csi.x-k8s.io
+  assert_success
+}
+
+@test "secretsync crd is established" {
+  kubectl wait --for condition=established --timeout=60s crd/secretsyncs.secret-sync.x-k8s.io
+
+  run kubectl get crd/secretsyncs.secret-sync.x-k8s.io
+  assert_success
+}
+
+@test "Test rbac roles and role bindings exist" {
+  run kubectl get clusterrole/secrets-store-sync-controller-manager-role
+  assert_success
+
+  run kubectl get clusterrolebinding/secrets-store-sync-controller-manager-rolebinding
+  assert_success 
+}
+
+@test "[v1alpha1] deploy e2e-providerspc secretproviderclass crd" {
```

Comment:
What is the purpose of this test? We're doing the same set of steps again [here](https://github.com/kubernetes-sigs/secrets-store-sync-controller/pull/40/files#diff-395e5ad96d74d81648ec476082473348247fb32c3bbe242149b3c8b4144e7d85R43-R48).